Analysis/PhotonEnergyCorrect/src/photonCorrectionProcessor.cc

```diff
@@ -0,0 +1,204 @@
+#include "photonCorrectionProcessor.h"
+#include <iostream>
+#include <marlin/Global.h>
+#include "lcio.h"
+
+#include "IMPL/LCCollectionVec.h"
+#include "IMPL/ReconstructedParticleImpl.h"
+
+using namespace lcio ;
+using namespace marlin ;
+using std::cout;
+using std::endl;
+
+photonCorrectionProcessor aphotonCorrectionProcessor ;
+
+
+photonCorrectionProcessor::photonCorrectionProcessor() : Processor("photonCorrectionProcessor") {
+  // processor description
+  _description = "photonCorrectionProcessor applies an energy correction to photon-like PFOs" ;
+
+  registerProcessorParameter("inputCollection", "name of input PFO collection",
+                             _inputCollection, std::string("PandoraPFOs") );
+
+  registerProcessorParameter("useCorrectorDefaults" , "use defaults correction paramteres (as defined in photonCorrector", 
+			     _useCorrectorDefaults, true );
+
+  registerProcessorParameter("interModGaps_haveBeenCorrectedAtHitLevel", 
+			     "have the inter-Module gaps already been corrected at this hit level? (affects default param set used)",
+			     _interModGaps_haveBeenCorrectedAtHitLevel, true);
+
+  registerProcessorParameter("barrel_limit"               , "_barrel_limit               parameter",  _barrel_limit                , float( -999. ) );
```

Comment:
Is there a way to get meaningful descriptions for parameters ? Even though the energy correction seems tricky...

Reply:
I tried to improve the descriptions. README also now has more details.